rocketpy/motors/SolidMotor.py

```diff
@@ -0,0 +1,736 @@
+            is defined by the motor's axis of symmetry. The origin of the
+            coordinate system  may be placed anywhere along such axis, such as
+            at the nozzle area, and must be kept the same for all other
+            positions specified. Options are "nozzleToCombustionChamber" and
+            "combustionChamberToNozzle". Default is "nozzleToCombustionChamber".
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(
+            thrustSource,
+            dry_mass,
+            center_of_dry_mass,
+            dry_inertia,
+            nozzleRadius,
+            burn_time,
+            nozzlePosition,
+            reshapeThrustCurve,
+            interpolationMethod,
+            coordinateSystemOrientation,
+        )
+        # Nozzle parameters
+        self.throatRadius = throatRadius
+        self.throatArea = np.pi * throatRadius**2
+
+        # Grain parameters
+        self.grainsCenterOfMassPosition = grainsCenterOfMassPosition
+        self.grainNumber = grainNumber
+        self.grainSeparation = grainSeparation
+        self.grainDensity = grainDensity
+        self.grainOuterRadius = grainOuterRadius
+        self.grainInitialInnerRadius = grainInitialInnerRadius
+        self.grainInitialHeight = grainInitialHeight
+
+        # Grains initial geometrical parameters
+        self.grainInitialVolume = (
+            self.grainInitialHeight
+            * np.pi
+            * (self.grainOuterRadius**2 - self.grainInitialInnerRadius**2)
+        )
+        self.grainInitialMass = self.grainDensity * self.grainInitialVolume
+
+        self.evaluateGeometry()
+        return None
+
+    @funcify_method("Time (s)", "Mass (kg)")
+    def propellantMass(self):
+        """Evaluates the total propellant mass as a function of time.
+
+        Returns
+        -------
+        Function
+            Mass of the motor, in kg.
+        """
+        return self.grainVolume * self.grainDensity * self.grainNumber
+
+    @funcify_method("Time (s)", "Grain volume (m³)")
+    def grainVolume(self):
+        """Evaluates the total propellant volume as a function of time. The
+        propellant is assumed to be a cylindrical Bates grain under uniform
+        burn.
+
+        Returns
+        -------
+        Function
+            Propellant volume as a function of time.
+        """
+        cross_section_area = np.pi * (
+            self.grainOuterRadius**2 - self.grainInnerRadius**2
+        )
+        return cross_section_area * self.grainHeight
+
+    @funcify_method("Time (s)", "Exhaust velocity (m/s)")
+    def exhaustVelocity(self):
+        """Exhaust velocity by assuming it as a constant. The formula used is
+        total impulse/propellant initial mass.
+
+        Returns
+        -------
+        self.exhaustVelocity : rocketpy.Function
+            Gas exhaust velocity of the motor.
+        """
+        return self.totalImpulse / self.propellantInitialMass
```

Comment:
Even if it a constant value, I think you should return a rocketpy.Function object.

This would improve the consistency with other motor classes

Reply:
The same comment that I made for your earlier suggestion applies here. The value is constant, but is turned into a Function by funcify.

Explicitly calling the Function is a bit redundant.